src/filter_iterator.c

```diff
@@ -803,8 +803,8 @@ EnrichedChunk *AggregationIterator_GetNextChunk(struct AbstractIterator *iter) {
                 // - mod where 0 <= mod from (1)+(2) contextScope > chunk->samples.timestamps[0]
                 // from (3) chunk->samples.timestamps[0] >= self->aggregationLastTimestamp so the
                 // following condition should always be false on the first iteration
-                if ((is_reversed == FALSE && sample.timestamp >= contextScope) ||
-                    (is_reversed == TRUE && sample.timestamp < self->aggregationLastTimestamp)) {
+                if ((is_reversed == false && sample.timestamp >= contextScope) ||
```

Comment:
Same: don't compare to true or false. Just use the values or their negations.

Reply:
I feel like this can be DeMorgan'd. but it's not worth the aggrivation.

Reply:
Not sure the logic will be clearer (and anyway optimizers know to DeMorganize; maybe even Karnaugh-map-ize? who knows).

Reply:
of course if i thought the optimizer couldn't (or wouldn't), then it would have been worth the aggravation.